drivers/hmis/app/models/hmis/ce/match/expression/homeless_days_calculator.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Hmis::Ce::Match::Expression
+  # Calculates the total number of days a client has been homeless.
+  # Excludes days when the client was housed (housed status takes precedence over homeless).
+  class HomelessDaysCalculator
+    def initialize(current_date)
+      @current_date = current_date
+    end
+
+    def call(clients)
+      client_ids = clients.pluck(:id)
+      housed_dates_by_client = fetch_housed_dates(client_ids)
+      homeless_dates_by_client = fetch_homeless_dates(client_ids)
+
+      calculate_homeless_days(client_ids, housed_dates_by_client, homeless_dates_by_client)
+    end
+
+    private
+
+    def fetch_housed_dates(client_ids)
+      # First identify any days the client was housed, as these will be excluded from the count.
+      # A client can have both homeless and housed service history records on the same day,
+      # with "housed" taking precedence.
+      housed_dates = GrdaWarehouse::ServiceHistoryService.non_homeless.
```

Comment:
should we continue to rely on ServiceHistoryService to calculate days homeless?

Reply:
We don't have a use-case for a days homeless calculation in the MVP CE implementation, so I would advocate for not spending time on it at this point. I think we can remove it altogether and create a ticket for implementing when needed.

Reply:
Works for me!

Reply:
It's captured in https://github.com/open-path/Green-River/issues/7840

Reply:
removed Days homeless